src/com/googlecode/jmxtrans/model/output/GraphiteWriter.java

```diff
@@ -101,6 +101,11 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 							log.debug("Graphite Message: {}", line);
 							writer.write(line);
 							writer.flush();
```

Comment:
The real question is performance vs data loss. If we checkError / flush for each metric, we ensure that problems are detected early and corrected as soon as possible. If we check late, we ensure optimal use of buffers and less pressure on network. I would tend to go with performance instead of consistency. I think loosing a few data point is OK if we can ensure high throughput.